### ISSUE TITLE
Debugging support

### ISSUE THREAD

**Issue by valeros** [platformio/platform-shakti#1]
Hi @sathya281! Could you please describe (or ask someone who knows) how you debug your boards at the moment? What software, debug probe, target do you use (for example `artix7_100t` board)?

**Comment by ghost** [comment]
Hi,

First of all thanks for the platform support.

Hope below answer helps you:

Board : artix7_100t and artix7_35t
debug probe: ftdi interface connected through riscv openocd
Debugger: riscv gnu GDB, 32 bit / 64 bit


For openocd, we want to use below commit. This has ftdi support.
https://github.com/riscv/riscv-openocd/tree/fd9c54b1feea13f6846c69fc9e1e8dc3a7879717
 

regards,
Sathya

**Comment by valeros** [comment]
Could you post here a simple command for openocd with all arguments that you are using?

**Comment by ghost** [comment]
command -
**sudo openocd -f spike.cfg**

**contents of spike.cfg**
#interface_list
interface jlink
debug_level 2 

#interface remote_bitbang
#interface usb_blaster

#remote_bitbang_host localhost
#remote_bitbang_port 10000


set _CHIPNAME riscv
jtag newtap $_CHIPNAME cpu -irlen 5 -expected-id 0x100039d3
#jtag newtap $_CHIPNAME cpu -irlen 5 -expected-id 0x10e31913

set _TARGETNAME $_CHIPNAME.cpu
target create $_TARGETNAME riscv -chain-position $_TARGETNAME

adapter_khz 900 
#$_TARGETNAME configure -event reset-start { adapter_khz 200 }
#riscv set_prefer_sba on
gdb_report_data_abort enable

init
halt

**Comment by ghost** [comment]
This one is for ftdi stuff
command -
**sudo openocd -f spike.cfg**

**contents of spike.cfg**
debug_level 1
adapter_khz 8000
interface jlink
gdb_port 3333
telnet_port 4444
tcl_port 6666

set _CHIPNAME riscv
jtag newtap $_CHIPNAME cpu -irlen 5 -expected-id 0x100039D3

set _TARGETNAME $_CHIPNAME.cpu
target create $_TARGETNAME riscv -chain-position $_TARGETNAME

gdb_report_data_abort enable

riscv set_prefer_sba on

init

**Comment by valeros** [comment]
In both scripts, `jlink` is specified as the interface: `interface jlink`. Is that right?

**Comment by ghost** [comment]
Hey sorry, Thanks for pointing it out.

I have listed below the config files for both the debug probe.

1. FTDI support

**sudo openocd -f ftdi_spike.cfg**
[ftdi_spike.cfg](https://gitlab.com/shaktiproject/cores/shakti-soc/blob/master/fpga/boards/artya7-100t/e-class/shakti-arty.cfg)

2. JLINK support 

**sudo openocd -f jlink_spike.cfg**
[jlink_spike.cfg](https://gitlab.com/shaktiproject/cores/shakti-soc/blob/master/fpga/boards/artya7-100t/e-class/shakti-jlink-jtag.cfg)

**Comment by valeros** [comment]
Any plans for adding those scripts to the official `shakti-sdk` repository? At the moment there is only one `spike.cfg` for jlink debug probe.

**Comment by ghost** [comment]
We are working on it. In 24hrs, we will add.

**Comment by ghost** [comment]
Hi,
you can check now. The SDK is updated.

**Comment by ghost** [comment]
I went through the code. We need to change lots of data in JSON file

**Comment by valeros** [comment]
Sure thing, feel free to submit a PR.

**Comment by johnnavaj** [comment]
@valeros how do i pull /use this in pio ? i don't see any entry added for the boards mentioned.

**Comment by valeros** [comment]
You need to install this platform from the git repository, in case you are using the command line version:
```
pio platform install https://github.com/platformio/platform-shakti.git
```
or `PIO Home -> Platforms -> Advanced Installation` in the IDE

**Comment by johnnavaj** [comment]
Yes i was able to load the platform into the ide , as part of the board entries i don't find the boards as part of shakti-platform.

**Comment by valeros** [comment]
I'm not sure I understand your question. Can you please describe in detail what you're trying to achieve?

**Comment by johnnavaj** [comment]
I was trying to understand how to add the board entry (ie), when you create a project it should be able to point to the designated board which we have support for eg - artix7_100t or artix7_35t also along with the SDK. 

So i discussed with @sathya281 , i will make the changes and submit a PR soon.
